app/services/teacher_training_public_api/sync_courses.rb

```diff
@@ -12,28 +12,97 @@ def perform(provider_id, recruitment_cycle_year)
         year: recruitment_cycle_year,
         provider_code: @provider.code,
       ).paginate(per_page: 500).each do |course_from_api|
-        update_course(course_from_api, recruitment_cycle_year)
+        create_or_update_course(course_from_api, recruitment_cycle_year)
       end
     rescue JsonApiClient::Errors::ApiError
       raise TeacherTrainingPublicAPI::SyncError
     end
 
   private
 
-    # As we’re augmenting existing data, don’t create fresh incomplete courses,
-    # only update those that already exist
-    def update_course(course_from_api, recruitment_cycle_year)
-      course = provider.courses.find_by(
+    def create_or_update_course(course_from_api, recruitment_cycle_year)
+      course = provider.courses.find_or_create_by(
         code: course_from_api.code,
         recruitment_cycle_year: recruitment_cycle_year,
-      )
+      ) do |new_course|
+        new_course.open_on_apply = new_course.in_previous_cycle&.open_on_apply ? true : false
```

Comment:
Sorry if I'm being daft, but why are we checking the previous cycles attr?

Reply:
I'm not sure I understand what you mean. I copied this logic over from the old job.

Reply:
The first time we learn about a course we check whether it was open last cycle, and if so, automatically open it. This saved the transition team having to go and manually reopen all the already-open courses at the start of the new cycle.